ModelViewIntentSample/core/src/main/java/com/novoda/movies/mvi/search/BaseStore.kt

```diff
@@ -2,34 +2,38 @@ package com.novoda.movies.mvi.search
 
 import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.disposables.Disposable
-import io.reactivex.functions.BiFunction
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.PublishSubject
 
 class BaseStore<A, S, C>(
     private val schedulingStrategy: SchedulingStrategy,
     private val reducer: Reducer<S, C>,
     private val middlewares: List<Middleware<A, S, C>>,
-    initialValue: S
+    private val initialValue: S
 ) : Store<A, S, C> {
-    private val changes = BehaviorSubject.create<C>()
+    private val changes = PublishSubject.create<C>()
     private val state = BehaviorSubject.createDefault(initialValue)
     private val actions: PublishSubject<A> = PublishSubject.create()
 
     override fun wire(): Disposable {
         val disposables = CompositeDisposable()
-        val newState = changes.withLatestFrom(state, BiFunction<C, S, S> { change, state ->
-            reducer.reduce(state, change)
-        })
-        disposables.add(newState
+
+        val newState = changes.scan(
```

Comment:
Can you give a bit of context on this change? 
Why is scan preferred over the previous approach?

Reply:
The main benefit is that it doesn't require us consuming the previous state from another observable since the state now _lives in the pipeline_. 
So when there is a `new change` event, the scan operator will provide us the `previous state`.